### PR TITLE
[Hub Generated] Review request for Microsoft.VirtualMachineImages to add version stable/2022-02-14

### PR DESCRIPTION
**Pull request by zdelacerda-microsoft**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/zdelacerda-microsoft/azure-rest-api-specs/imagebuilder-Microsoft.VirtualMachineImages-2022-02-14...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (sumit-kalra) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1232]: Have you tried building model for AIB? If these Swagger changes lead to any major changes in our model then there will be more work to fix the model. It would be nice to know what kind of change in the model this Swagger change will cause.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (sumit-kalra) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json]: If "#" was present in the ADO version then how was auto-rest still able to find the ProxyResource and generate correct model for us. Removing "#" is not an incorrect thing to do but I am just curios how our model worked with "#" in the path.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (sumit-kalra) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: From documentation (http://azure.github.io/autorest/extensions/#x-ms-identifiers), this extension seems to be "InProgress" and not ready. So why are we adding this?

Comment (sumit-kalra) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: Why is x-ms-identifier added only for "inVMValidations" array but not for other arrays - like "distribute"?
Regardless, I would much rather prefer removing it from here rather than adding everywhere else - to reduce churn.

Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: this change was made because it was causing the swagger lintdiff check to fail. the error is no longer showing as it's now been resolved but this is what it looked like:
83293e8a-a951-410c-ae86-dc092d686a68

Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: i added this as it was causing the lintdiff check to fail. perhaps @honghr can help answer this question.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json]: the spec file we use to generate datamodels in ADO actually uses a different path than what is seen above since it's stored in a different directory (see swagger/api/2022-02-14/imagebuilder.server.json).

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (feiyushi) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: see this comment: https://github.com/Azure/azure-rest-api-specs/pull/18896#issuecomment-1116746656
I think the other places are not asked to be fixed since they were introduced before this rule was in place.
From the swagger model definition i don't think we have an identifier for those array objects. empty  x-ms-identifiers may be correct. @sumit-kalra thoughts?

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (feiyushi) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json]: https://swagger.io/docs/specification/using-ref/
```
Local Reference – $ref: '#/definitions/myElement' # means go to the root of the current document and then find elements definitions and myElement one after one.
```

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (rkmanda) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:624]: >This image must reside in the same subscription and region  [](http://example.com/codeflow?start=98&length=59)

Is this a new restriction being introduced with this API version or is this just an update to the description?

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (sumit-kalra) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:624]: This was always a requirement that was enforced but not documented here. We are just clarifying the documentation here.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (rkmanda) [inline specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:845]: >int32 [](http://example.com/codeflow?start=21&length=5)

waiting for breaking changes approval before signing off. Rest of the ARM changes look fine.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: I have not tried building the model for AIB yet. Once we get ARM and breaking change approval I will generate a new model.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json]: This was removed by request of breaking change reviewer, however it is triggering a failure in Swagger LintDiff tests. Awaiting breaking change reviewer to approve this removal.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (honghr) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: Hi @sumit-kalra @zdelacerda-microsoft The extension is ready now, but the doc may have some confusion. We will ask the author to fix it. Thanks.

**Comment by jorgecotillo** [comment]
Signed off from ARM side since seems like the spec was already reviewed.

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1232]: as of commit #11. i have copied the changes from this PR into our AIB and have no problems with datamodel (there are no changes to it)

File: specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json
Comment (zdelacerda-microsoft) [reply specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2022-02-14/imagebuilder.json:1229]: the extension being inProgress seems to be an issue with documentation according to honghr's comment. i think it might be better to keep it simple and only add this where the linting error indicates. from my experience these requirements are always changing and it might be something different during the next api release. it may be redundant to add this everywhere and then have to remove it in the future.

**Review by honghr — APPROVED** [review]

**Comment by zdelacerda-microsoft** [comment]
@rkmanda do you have any unresolved comments on this PR? I see you requested changes but the comment is empty.

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**
